modules/page-edit.php

```diff
@@ -108,14 +148,33 @@
 			{
 				$content .= "<p><strong>Warning: You are not logged in! Your IP address <em>may</em> be recorded.</strong></p>";
 			}
-			$content .= "<form method='post' action='index.php?action=save&page=" . rawurlencode($page) . "&action=save' class='editform'>
-			<input type='hidden' name='prev-content-hash' value='" . sha1($pagetext) . "' />
-			<textarea name='content' autofocus tabindex='1'>$pagetext</textarea>
-			<pre class='fit-text-mirror'></pre>
-			<input type='text' name='tags' value='$page_tags' placeholder='Enter some tags for the page here. Separate them with commas.' title='Enter some tags for the page here. Separate them with commas.' tabindex='2' />
-			<p class='editing-message'>$settings->editing_message</p>
-			<input name='submit-edit' type='submit' value='Save Page' tabindex='3' />
-		</form>";
+			
+			// Include preview, if set
+			if(isset($_POST['preview-edit']) && isset($_POST['content'])) {
+				// Need this for the prev-content-hash to prevent the conflict page from appearing
+				$old_pagetext = $pagetext;
+
+				// set the page content to the newly edited content
+				$pagetext = $_POST['content'];
+
+				// Set the tags to the new tags, if needed
+				if(isset($_POST['tags']))
+					$page_tags = $_POST['tags'];
+
+				// Insert the "view" part of the page we're editing
+				$content .= parse_page_source($pagetext);
+
+			}
+
+			$content .= "<form method='post' name='edit-form' action='index.php?action=preview_edit&page=' class='editform'>
+					<input type='hidden' name='prev-content-hash' value='" . ((isset($old_pagetext)) ? sha1($old_pagetext) : sha1($pagetext)) . "' />
+					<textarea name='content' autofocus tabindex='1'>$pagetext</textarea>

```

Comment:
I don't think this is a security issue, since we're hashing it into a hexadecimal value before we put it on the page.  If we do wrap it in htmlentities() here, then it'd also need to be done in the save action as well, to prevent it from showing the conflicts page erroneously.